coprocess/grpc/services_test.go

```diff
@@ -0,0 +1,69 @@
+package grpc
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+
+	"github.com/TykTechnologies/tyk/config"
+	"github.com/TykTechnologies/tyk/coprocess"
+	"github.com/TykTechnologies/tyk/gateway"
+)
+
+func newTestGRPCServer() (s *grpc.Server) {
+	s = grpc.NewServer(
+		grpc.MaxRecvMsgSize(grpcTestMaxSize),
+		grpc.MaxSendMsgSize(grpcTestMaxSize),
+	)
+	coprocess.RegisterDispatcherServer(s, &dispatcher{})
+	return s
+}
+
+func startTestServices(tb testing.TB) (*gateway.Test, func()) {
+	tb.Helper()
+
+	listener, err := net.Listen("tcp", ":0")
+	require.NoError(tb, err)
```

Comment:
Previously the listener would use the static port and never close the socket, no error check.
Anyway. It runs on a random port now.